internal/daemon/controller/common/scopeids/scope_ids.go

```diff
@@ -353,3 +353,143 @@ func GetListingScopeIds(
 	}
 	return scopeResourceInfo.ScopeIds, scopeMap, nil
 }
+
+// AuthorizedForList retrieves and returns all scopes where a user is authorized
+// to perform a *list* action on. It looks recursively from `rootScopeId`.
+func AuthorizedForList(
+	ctx context.Context, // The context to use when listing in the DB, if required
+	iamRepoFn common.IamRepoFactory, // An IAM repo function to use for a scope listing call
+	authResults auth.VerifyResults, // The original auth results from the list command
+	rootScopeId string, // The scope ID to use, or the starting point for a recursive search
+	resourceType resource.Type, // The type of resource being listed
+	recursive bool, // Whether the search is recursive.
```

Comment:
I think we should drop this, and instead any calling code should just not call this function if it is a non-recursive call.

Reply:
Yep, addressed in [9fd0fe9](https://github.com/hashicorp/boundary/pull/2336/commits/9fd0fe983420986c11fe4364a2ff7a0862284608)